packages/flutter/lib/src/cupertino/form_section.dart

```diff
@@ -0,0 +1,316 @@
+// iOS 14.2 SDK.
+const EdgeInsetsDirectional _kDefaultInsetGroupedRowsMargin =
+    EdgeInsetsDirectional.fromSTEB(16.5, 0.0, 16.5, 16.5);
+
+// Used for iOS "Inset Grouped" border radius, estimated from SwiftUI's Forms in
+// iOS 14.2 SDK.
+// TODO(edrisian): This should be a rounded rectangle once that shape is added.
+const BorderRadius _kDefaultInsetGroupedBorderRadius =
+    BorderRadius.all(Radius.circular(10.0));
+
+// Used to differentiate the edge-to-edge section with the centered section.
+enum _CupertinoFormSectionType { base, insetGrouped }
+
+/// An iOS-style form section.
+///
+/// The base constructor for [CupertinoFormSection] constructs an
+/// edge-to-edge style section which includes an iOS-style header, footer, rows,
+/// the dividers between rows, and borders on top and bottom of the rows.
+///
+/// The [CupertinoFormSection.insetGrouped] constructor creates a round-edged and
+/// padded section that is commonly seen in notched-displays like iPhone X and
+/// beyond. Creates an iOS-style header, footer, rows, and the dividers
+/// between rows. Does not create borders on top and bottom of the rows.
+///
+/// The [header] parameter sets the form section header. The section header lies
+/// above the [children] rows, with margins that match the iOS style.
+///
+/// The [children] parameter is required and sets the list of rows shown in
+/// the section. The [children] parameter takes a list, as opposed to a more
+/// efficient builder function that lazy builds, because forms are intended to
+/// be short in row count. It is recommended that only [CupertinoFormRow] and
+/// [CupertinoTextFormFieldRow] widgets be included in the [children] list in
+/// order to retain the iOS look.
+///
+/// The [margin] parameter sets the spacing around the content area of the
+/// section encapsulating [children].
+///
+/// The [groupDecoration] parameter sets the decoration for the row group.
+/// If null, defaults to a conditional background color for the row group,
+/// with [CupertinoColors.systemBackground] in light themes and
+/// [CupertinoColors.secondarySystemBackground] in dark themes.
+/// If null, defaults to 10.0 circular radius when constructing with
+/// [CupertinoFormSection.insetGrouped]. Defaults to zero radius for the
+/// standard [CupertinoFormSection] constructor.
+///
+/// The [backgroundColor] parameter sets the background color for the section.
+/// If null, defaults to a conditional background color for the section, with
+/// [CupertinoColors.secondarySystemBackground] in light themes and
+/// [CupertinoColors.systemBackground] in dark themes.
+///
+/// {@macro flutter.material.Material.clipBehavior}
+class CupertinoFormSection extends StatelessWidget {
+  /// Creates a section that mimicks standard iOS forms.
+  ///
+  /// The base constructor for [CupertinoFormSection] constructs an
+  /// edge-to-edge style section which includes an iOS-style header, footer,
```

Comment:
We don't seem to have a `footer` field?

Reply:
I'll add it to the list of additions later on. @xster suggested we land this and look to more parameters like footer, etc. later.